service/frontend/service.go

```diff
@@ -198,6 +177,23 @@ func (s *Service) Start() {
 
 	// must start resource first
 	s.Resource.Start()
+
+	startCtx, cancel := context.WithTimeout(context.Background(), time.Second) // should take nearly no time at all
+	defer cancel()
+	if err := collections.user.OnStart(startCtx); err != nil {
+		logger.Fatal("failed to start user global ratelimiter collection", tag.Error(err))
+	}
+	if err := collections.worker.OnStart(startCtx); err != nil {
+		logger.Fatal("failed to start worker global ratelimiter collection", tag.Error(err))
+	}
+	if err := collections.visibility.OnStart(startCtx); err != nil {
+		logger.Fatal("failed to start visibility global ratelimiter collection", tag.Error(err))
+	}
+	if err := collections.async.OnStart(startCtx); err != nil {
+		logger.Fatal("failed to start async global ratelimiter collection", tag.Error(err))
+	}
+	cancel()
```

Comment:
feels more verbose than I'd like, and I don't like the `logger.Fatal`s, but:
- I'm not sure this is worth bundling together more nicely.  I'd rather push for adopting Fx more thoroughly.
- logger.Fatal is the normal way to handle errors during startup in these daemons :\